src/App.tsx

```diff
@@ -1,24 +1,213 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { Etodos, ResponseError } from './types/enum';
+import { UserData } from './types/userData';
+import { Todo } from './types/Todo';
+import {
+  createTodo,
+  deleteTodoOnServer,
+  getTodos,
+  updateTodoProp,
+} from './api';
+import { TodoItem } from './components/Todo';
+import { TodoHeader } from './components/TodoHeader';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
 
-const USER_ID = 0;
+const user: UserData = {
+  createdAt: '2023-07-10T13:09:53.578Z',
+  email: 'gookidoo@gmail.com',
+  id: 11038,
+  name: 'Віктор Булденко',
+  phone: null,
+  updatedAt: '2023-07-10T13:09:53.578Z',
+  username: null,
+  website: null,
+};
+
+let isUncomplete = 0;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isShowFooter, setIsShowFooter] = useState<boolean>(true);
```

Comment:
No need in state. You can simple create variable inside component and use it:
```
const hasFooter = !!todos.length
```

Reply:
no, we can`t, because what happens if active or completed todos where sum 0? Yeah, right - the footer was unmounted, but we still need the footer mounted. It's bad advice